assembly/integer/i128.ts

```diff
@@ -296,18 +299,23 @@ export class i128 {
 
   @inline @operator('*')
   static mul(a: i128, b: i128): i128 {
-    var ah = i128.shr(a, 64).i128ToU64();
-    var al = a.i128ToU64();
-    var bh = i128.shr(b, 64).i128ToU64();
-    var bl = b.i128ToU64();
-    var lo = al * bl;
-    var mid1 = ah * bl;
-    var mid2 = al * bh;
-    var hi = ah * bh;
-    var carry = u64((lo >>> 32) + (mid1 & 0xffffffff) + (mid2 & 0xffffffff));
-    var lo64 = u64(lo) + (u64(mid1) << 32) + (u64(mid2) << 32);
-    var hi64 = hi + (carry >>> 32) + ((mid1 >>> 32) + (mid2 >>> 32));
-    return new i128((hi64 << 64) | lo64);
+    if (b == i128.One) {
+      return a;
+    }
+    var neg = false;
+    if (a < i128.Zero) {
+      a = a.neg();
+      neg = !neg;
+    }
+    if (b < i128.Zero) {
+      b = b.neg();
+      neg = !neg;
+    }
+    var prod = new i128(__multi3(a.lo, a.hi, b.lo, b.hi),__res128_hi);
+    if (neg) {
+      prod = prod.neg();
+    }
+    return prod;
```

Comment:
You don't need handle sign for multiply. Just directly use `i128(__multi3(a.lo, a.hi, b.lo, b.hi),__res128_hi)`. Multiplication is some-kind of sign-agnostic operation, unlike of division.